### DISCUSSION TITLE
How to move the UI to the right monitor on multi-monitor?

### DISCUSSION THREAD

**Discussion by KSXGitHub** [davatorium/rofi#1804] [(Possible) Issues] [answered]
I use rofi on GNOME Shell with multiple monitor. Whenever I open rofi, it always appear on whichever monitor the mouse cursor is on. Normally, I can just press <kbd>Super+Shift+Left/Right</kbd> to move application windows to the right monitor, but I can't do so with rofi.

My questions:
* How to choose the right monitor to open rofi on before launching? (Some kind of CLI flags)
* How to move rofi UI to the right monitor after it appears on the wrong monitor?

**Comment by DaveDavenport** [comment, marked as answer]
From manpage:
```markdown
-monitor name

Select monitor to display rofi on. It accepts as input: primary (if primary output is set), the xrandr output name, or integer number (in order of detection). Negative numbers are handled differently:

    -1: the currently focused monitor.
    -2: the currently focused window (that is, rofi will be displayed on top of the focused window).
    -3: Position of mouse (overrides the location setting to get normal context menu behavior.)
    -4: the monitor with the focused window.

    -5: the monitor that shows the mouse pointer.

    Default: -5

See rofi -h output for the detected monitors, their position, and size.
```

Not all options might work, this depends if the window manager sets the right hint.

If you use rofi on wayland, we don't support wayland.  If you use the rofi wayland port, please ask  the port maintainer.

**Reply by DaveDavenport** [reply]
You cannot move rofi once shown.

**Reply by KSXGitHub** [reply]
Interesting, `man rofi | grep monitor` shows result but `rofi -help | grep monitor` doesn't.

> You cannot move rofi once shown.

Is this intended? I would like to create an issue regarding this feature.

**Reply by DaveDavenport** [reply]
`-h`  shows the alias (-m)  doing a grep -i would have matched.

```
        -m [string]                             Monitor id to show on
```

Still the manpage is the place to be.


Yes. the type of window rofi makes it an "unmanaged" window (by the window manager) that overlaps everything.  Generic window manager keybindings don't work on it. Rofi also grabs all key input.

If you check the manpage, there is an option to make rofi behave more like a normal window.
(-normal-window), but in this way the window manager probably does not let the window work the way you expect.. 
(focus stealing prevention might stop it from grabbing focus,  it might be tiled/stacked differently as expected, etc.)
